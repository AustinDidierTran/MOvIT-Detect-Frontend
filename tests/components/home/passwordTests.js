import Enzyme, { mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import Password from '../../../src/components/home/password';

Enzyme.configure({ adapter: new Adapter() });

describe('Password Tests', () => {
  let wrapper;
  const initialState = {
    applicationReducer: { language: 'en' },
  };
  const submitSpy = sinon.spy();
  const forgotPasswordSpy = sinon.spy();
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    failed: false,
    onSubmit: (value) => { submitSpy(value); },
    onForgotPassword: () => { forgotPasswordSpy(); },
  };

  beforeEach(() => {
    submitSpy.resetHistory();
    forgotPasswordSpy.resetHistory();

    wrapper = mount(<Password store={store} {...props} />);
  });

  it('should have proptypes', () => {
    const actualValue = Password.WrappedComponent.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      onSubmit: PropTypes.func.isRequired,
      failed: PropTypes.bool.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should call onSumbit on login button pressed', () => {
    wrapper.setState({ password: 'test' });

    wrapper.find('#loginBtn').simulate('click');

    expect(submitSpy.calledOnce).toEqual(true);
  });

  it('should call onSumbit on login enter pressed', () => {
    wrapper.setState({ password: 'test' });

    wrapper.find('#password').simulate('keyPress', { keyCode: 13, key: 'Enter' });

    expect(submitSpy.calledOnce).toEqual(true);
  });

  it('should call reset password method', () => {
    wrapper.find('#forgotPasswordBtn').simulate('click');

    expect(forgotPasswordSpy.calledOnce).toEqual(true);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should macth snapshot when password failed', () => {
    const failedProps = {
      failed: true,
      onSubmit: (value) => { submitSpy(value); },
      onForgotPassword: () => { forgotPasswordSpy(); },
    };
    const passwordFailedWrapper = mount(<Password store={store} {...failedProps} />);

    expect(toJson(passwordFailedWrapper)).toMatchSnapshot();
  });
});
