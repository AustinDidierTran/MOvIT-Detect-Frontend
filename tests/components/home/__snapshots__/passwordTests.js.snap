// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Tests should macth snapshot when password failed 1`] = `
<div
  className="row"
>
  <div
    className="ui-inputgroup col-8 offset-2"
  >
    <input
      className="form-control"
      id="password"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="login.password.en"
      style={
        Object {
          "borderColor": "red",
        }
      }
      type="password"
      value=""
    />
    <button
      className="ui-button-secondary"
      id="loginBtn"
      onClick={[Function]}
    >
      <i
        className="fa fa-sign-in"
      />
    </button>
  </div>
  <div
    className="ui-inputgroup col-8 offset-2"
  >
    <button
      className="btn btn-link pl-0"
      id="forgotPasswordBtn"
      onClick={[Function]}
      type="button"
    >
      login.forgotPassword.en
    </button>
  </div>
  <div
    className="row"
    style={
      Object {
        "marginTop": "2px",
      }
    }
  >
    <div
      className="col-8 offset-2"
    >
      <Message
        className={null}
        id="errorMsg"
        severity="error"
        style={null}
        text="login.wrongPassword.en"
      />
    </div>
  </div>
</div>
`;

exports[`Password Tests should match the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="ui-inputgroup col-8 offset-2"
  >
    <input
      className="form-control"
      id="password"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="login.password.en"
      style={Object {}}
      type="password"
      value=""
    />
    <button
      className="ui-button-secondary"
      id="loginBtn"
      onClick={[Function]}
    >
      <i
        className="fa fa-sign-in"
      />
    </button>
  </div>
  <div
    className="ui-inputgroup col-8 offset-2"
  >
    <button
      className="btn btn-link pl-0"
      id="forgotPasswordBtn"
      onClick={[Function]}
      type="button"
    >
      login.forgotPassword.en
    </button>
  </div>
</div>
`;
